src/combinatorics/simple_graph/clique.lean

```diff
@@ -0,0 +1,42 @@
+/-
+Copyright (c) 2022 Yaël Dillies, Bhavik Mehta. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yaël Dillies, Bhavik Mehta
+-/
+import combinatorics.simple_graph.basic
+import data.finset.pairwise
+
+/-!
+# Graph cliques
+
+This file defines cliques in simple graphs. A `n`-clique is a set of `n` vertices which are
+pairwise connected.
+
+## Main declarations
+
+* `simple_graph.is_n_clique`: Predicate for a set of vertices to be a `n`-clique.
+-/
+
+open finset fintype
+
+namespace simple_graph
+variables {α : Type*} (G H : simple_graph α) {n : ℕ} {s : finset α}
+
+/-- A `n`-clique in a graph is a set of `n` vertices which are pairwise connected. -/
+structure is_n_clique (n : ℕ) (s : finset α) : Prop :=
+(card_eq : s.card = n)
+(pairwise : (s : set α).pairwise G.adj)
```

Comment:
My idea is that `s.pairwise G.adj` is *already* good enough of a definition for `is_clique`, hence my code is already in the state you ask for in your second snippet. I can add the separate definition if you insist, though.

Reply:
I'd normally agree with you about how the short expression is its own definition, but I'm imagining that there will be dot notation that's useful to have that we wouldn't want to put on `set.pairwise`.

Reply:
Rereading your original comment, what you seem to want is the *type* of cliques, not a predicate for a set to be a clique. Or am I misinterpreting?

Either way, I'd be happy to defer until we actually need it. This is graph theory after all 😁.

Reply:
I really do mean a predicate on cliques. This way if `h : G.is_clique s`, we can write, say, `h.card_le_deg hv` where `hv : v ∈ s` to get `nat.card s ≤ G.deg v`.  Using a to-be-backported Lean 4 feature, if `h : G.is_n_clique n s`, then we will also be able to write `h.card_le_deg hv` for this. It's not the best example (since we'd want the lemma to be specialized to use `n` for `nat.card s`), but it was the first that came to mind to at least illustrate my thinking here.

Reply:
Another example could be `h.emb : complete_graph s ↪g G`

Reply:
I think being able to turn a clique into an embedding from a complete graph would be valuable, and the reverse direction: you can make this a TODO if you prefer